### PR TITLE
Adjust ATCameraDewar alarm range for the CCD temperature, and the hys…

### PR DESCRIPTION
**Pull request by patrickingraham**
…teresis value which was too large and was not allowing the alarm to shut off.

### CODE REVIEW

**Comment by craiglagegit** [comment]
Are we sure this fixes the problem?  It seems to me the problem is not that the limits were wrong, but that the logic is wrong.  If I read correctly, the old limits were -93 and -95.  The temperature was running at -94.16 and never went outside of these limits, so why were there errors?  I'm going from memory, but I think the error message said, "Temp of -94.16 exceeds limit of -95.0 + 1.0" or something like that.  Why the +1.0?  Is the 1.0 the number that was changed on line 281?  Shouldn't it be -1.0 on the lower side and +1.0 on the upper side?  Or am I confused?  In any case I think we need more verification that this will fix the problem.

**Comment by patrickingraham** [comment]
Sorry, I put this discussion in https://jira.lsstcorp.org/browse/SITCOM-419 but maybe you didn't see it?
See here for why the issue is with the hysteresis value:
https://jira.lsstcorp.org/browse/SITCOM-419?focusedCommentId=724013&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-724013

That being said, we could open the limits a little wider if you want.

**Comment by craiglagegit** [comment]
OK, I read your note, but I'm still lost.  I don't understand what "The reason for the alarm was that it had not surpassed the appropriate amount of "hysteresis" from the alarm limit." means.
The temperature virtually never went outside of (-94.0, -94.35).  So why exactly did it alarm?
________________________________

**Comment by patrickingraham** [comment]
The limit was at -95, with a hysteresis value of 1, meaning it had to go to -94, and stay above that value for several minutes, to clear the alarm. This hysteresis value is used to prevent "fluttering" of alarms when things are near the limit.
However, it was never going to do that because the setpoint was -94.16.

**Comment by patrickingraham** [comment]
no that's the hysteresis value... so that means that if the detector gets too cold, then the temperature will have to get to -95.16+0.75 = -94.51 and then the alarm will clear after 3 minutes.

**Comment by craiglagegit** [comment]
OK, I think I see.  I'll approve it now.  Thanks for explaining.

________________________________

**Comment by craiglagegit** [comment]
OK, I think I understand now.  I think this will fix the problem, so I am approving.

**Comment by craiglagegit** [comment]
I see that I am not a reviewer, but I agree with the changes, FWIW.

File: Jenkinsfile [hunk 1]
```diff
@@ -116,8 +116,8 @@ pipeline {
     post {
         always {
             // Change ownership of the workspace to Jenkins for clean up.
-            withEnv(["HOME=${env.WORKSPACE}"]) {
-                sh 'chown -R 1003:1003 ${HOME}/'
+            withEnv(["WHOME=${env.WORKSPACE}"]) {
+                sh 'chown -R 1003:1003 ${WHOME}/'
```
Comment (r-owen) [inline Jenkinsfile:120]: I believe this should be left as `HOME` (only in this final cleanup section). That is certainly what I was told and what I have done in my other packages (as I get to them).

**Review by r-owen — CHANGES_REQUESTED** [review]
Thank you for updating the Jenkinsfile, but I think the changes in the post section may need to be undone (though I'd be thrilled if that was not the case -- it's easier to use WHOME everywhere).

The changes to the default values of the ATCameraDewar rule look good to me.

**Comment by r-owen** [comment]
> OK, I think I understand. But looking at the pull request, I don't see the change to the hysteresis value. Is it somewhere else? I see line 281, where "The amount by which a value must improve past an alarm threshold before the alarm severity decreases." is changed from 1.0 to 0.75. Is that something else?
> […](#)

If you expand the text a bit in the diff you will see that the parameter you mentioned is named "temperature_hysteresis".

File: Jenkinsfile (hunk 1 above)
Comment (patrickingraham) [reply Jenkinsfile]: OK... I don't really understand why this is but that's ok. I thought it was a global change.
Thanks for the catch.

**Review by r-owen — APPROVED** [review]

**Pull request merged**